docs/architecture/specs/nft/README.md

```diff
@@ -0,0 +1,291 @@
+
+
+### NFT Specs (via ERC-1155)
+
+Nowadays the main standard for providing NFTs functionality in Ethereum networks is ERC-721. ERC-721 require a separate contract to be deployed for each token type or collection. This places a lot of redundant bytecode on the Ethereum blockchain and limits certain functionality by the nature of separating each token contract into its own permissioned address. This means high cost, complexity, etc.
+
+Additionaly to ERC-721, ERC-1155 implements a multi-token factory allowing to register and tokenize multiple and independent assets in a same contract instance without multiple deployments. ERC-1155 design permits transferring multiple token types at once, saving on transaction costs.It is also easy to describe and mix multiple fungible or non-fungible token types in a single contract.
+
+Because this advantages the following NFT integration will be based in the ERC-1155 foundations.
+
+## Architecture
+
+### NFTs Principles
+
+The NFTs engine is based in the following principles:
+
+* Every asset has a Decentralized Identifier (aka DID) attached, and every DID as a identifier representing a digital entity can have associated multiple NFTs, allowing the tokenization of any kind of digital asset independently of the physical asset behind of it
+* The user registering a DID can decide if wants to enable or not the tokenization of the asset via NFTs
+* When the tokenization is enabled for a specific asset, the user registering the asset can define a minting cap. This minting cap can not be changed afterward, because modifying the number of existing items of an asset will affect the further value of them for the NFTs holders. If the minting cap is set to zero, it means the DID minting is uncapped.
+* The user registering an asset can specify the royalties that are rewarding the original creator in the secondary market. This royalties must be between 0 and 100 percent. The royalties can not be changed after they are initialized. This protects to the buyers of a NFT to have to pay for a different commission to the one agreed during the purchase of a NFT.
+* The payment and transfer of NFTs must always respect the original creators attribution and rewards
+* Users giving or selling NFTs can have a mechanism to facilitate exclusive services to NFT holders
+
+
+### Implementation
+
+From a Smart Contracts point of view, the `DIDRegistry` now extends a the new `NFTUpgradeable` smart contract. This new contract implements the `ERC-1155` standard and it's based in OpenZeppelin implementation.
+
+With this change, when a new Asset is registered via the DIDRegistry, it can automatically `mint`, `burn` and `transfer` NFTs attached to the Asset. Example:
+
+```js
+
+await didRegistry.registerMintableDID(
+                did, checksum, [], url, cappedAmount, royalties, constants.activities.GENERATED, '')
+await didRegistry.mint(did, 5)
```

Comment:
Does this mean that I can mint multiple times as long I don't go over the cap?

Reply:
yes, you can mint as you go till you reach the total cap